fenzo-core/src/main/java/com/netflix/fenzo/InternalVMCloner.java

```diff
@@ -42,38 +43,45 @@
         final Map<String, Double> scalars = new HashMap<>();
         final Map<String, Protos.Attribute> attributeMap = new HashMap<>();
         if (avms != null) {
-            for(AssignableVirtualMachine avm: avms) {
-                final Map<VMResource, Double[]> resourceStatus = avm.getResourceStatus();
-                Double[] values = resourceStatus.get(VMResource.CPU);
-                if (values != null && values.length == 2)
-                    cpus = Math.max(cpus, values[0] + values[1]);
-                values = resourceStatus.get(VMResource.Memory);
-                if (values != null && values.length == 2)
-                    mem = Math.max(mem, values[0] + values[1]);
-                values = resourceStatus.get(VMResource.Disk);
-                if (values != null && values.length == 2)
-                    disk = Math.max(disk, values[0] + values[1]);
-                values = resourceStatus.get(VMResource.Network);
-                if (values != null && values.length == 2)
-                    network = Math.max(network, values[0] + values[1]);
-                values = resourceStatus.get(VMResource.Ports);
-                if (values != null && values.length == 2)
-                    ports = Math.max(ports, values[0] + values[1]);
+            for (AssignableVirtualMachine avm : avms) {
+                Map<VMResource, Double> maxResources = avm.getMaxResources();
```

Comment:
we confirmed this is set because the pseudo scheduling happens after the main scheduling.